lib/livebook_web/live/session_live.ex

```diff
@@ -1425,6 +1471,47 @@ defmodule LivebookWeb.SessionLive do
     )
   end
 
+  defp prune_outputs_in_data(data) do
+    %{
+      data
+      | notebook:
+          Notebook.update_cells(data.notebook, fn
+            %Cell.Elixir{} = cell ->
+              %{cell | outputs: prune_outputs(cell.outputs)}
+
+            cell ->
+              cell
+          end)
+    }
+  end
+
+  defp prune_outputs(outputs) do
+    outputs
+    |> Enum.reverse()
+    |> do_prune_outputs()
+    |> Enum.reverse()
+  end
+
+  defp do_prune_outputs([]), do: []
+
+  # Keep the last stdout, so that we know to message updates directly
+  defp do_prune_outputs([{idx, {:stdout, _}}]), do: [{idx, {:stdout, :__pruned__}}]
```

Comment:
Isn't this missing traverse the rest?

Like when you have a stdout after another output?

Reply:
This is reversed and we use body-recursion, so this is the head case.

Reply:
I see, perfect.